client/my-sites/stats/activity-log/index.jsx

```diff
@@ -193,27 +218,34 @@ class ActivityLog extends Component {
 			startDate,
 		} = this.props;
 
-		const startOfMonth = moment( startDate ).startOf( 'month' ),
-			startOfMonthMs = startOfMonth.valueOf(),
-			endOfMonthMs = moment( startDate ).endOf( 'month' ).valueOf();
-		const filteredLogs = filter( logs, obj => startOfMonthMs <= obj.ts_site && obj.ts_site <= endOfMonthMs );
-		const logsGroupedByDate = map(
+		const applySiteOffset = this.getSiteOffsetFunc();
+
+		// example: "2019 3"
+		const selectedMonthAndYear = moment( startDate ).format( 'YYYY M' );
+		const logsForMonth = filter( logs, ( { ts_utc } ) => {
+			return applySiteOffset( moment.utc( ts_utc ) ).format( 'YYYY M' ) === selectedMonthAndYear;
+		} );
+
+		const logsGroupedByDay = map(
 			groupBy(
-				filteredLogs,
-				log => moment( log.ts_site ).startOf( 'day' ).format( 'x' )
+				logsForMonth,
+				log => applySiteOffset( moment.utc( log.ts_utc ) ).format( 'LL' )
 			),
-			( daily_logs, timestamp ) => (
+			( daily_logs, day ) => (
 				<ActivityLogDay
 					allowRestore={ !! isPressable }
 					isRewindActive={ isRewindActive }
-					key={ timestamp }
+					key={ day }
 					logs={ daily_logs }
 					requestRestore={ this.handleRequestRestore }
 					siteId={ siteId }
-					timestamp={ +timestamp }
+					day={ day }
```

Comment:
Yeah I'm not certain on this, but I lean towards the already formatted string because it removes the question of whether or not the date needs to be offset before display.